docs/current/guides/snippets/azure-pipelines-container-instances/main.go

```diff
@@ -0,0 +1,133 @@
+	source := daggerClient.Host().Directory(".", dagger.HostDirectoryOpts{
+		Exclude: []string{"ci", "node_modules"},
+	})
+
+	// get Node image
+	node := daggerClient.Container(dagger.ContainerOpts{Platform: "linux/amd64"}).
+		From("node:18")
+
+	// mount source code directory into Node image
+	// install dependencies
+	// set entrypoint
+	c := node.WithDirectory("/src", source).
+		WithWorkdir("/src").
+		WithExec([]string{"cp", "-R", ".", "/home/node"}).
+		WithWorkdir("/home/node").
+		WithExec([]string{"npm", "install"}).
+		WithEntrypoint([]string{"npm", "start"})
+
+	// publish image
+	dockerHubUsername := os.Getenv("DOCKERHUB_USERNAME")
+	addr, err := c.WithRegistryAuth("docker.io", dockerHubUsername, dockerHubPassword).
+		Publish(ctx, fmt.Sprintf("%s/my-app", dockerHubUsername))
+	if err != nil {
+		panic(err)
+	}
+
+	// print ref
+	fmt.Println("Published at:", addr)
+
+	// initialize Azure credentials
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		panic(err)
+	}
+
+	// initialize Azure client
+	azureClient, err := armcontainerinstance.NewClientFactory(os.Getenv("AZURE_SUBSCRIPTION_ID"), cred, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	// define deployment request
+	containerGroup := armcontainerinstance.ContainerGroup{
+		Properties: &armcontainerinstance.ContainerGroupPropertiesProperties{
+			Containers: []*armcontainerinstance.Container{
+				{
+					Name: to.Ptr(containerName),
+					Properties: &armcontainerinstance.ContainerProperties{
+						Command:              []*string{},
+						EnvironmentVariables: []*armcontainerinstance.EnvironmentVariable{},
+						Image:                to.Ptr(addr),
+						Ports: []*armcontainerinstance.ContainerPort{
+							{
+								Port: to.Ptr[int32](3000),
+							}},
+						Resources: &armcontainerinstance.ResourceRequirements{
+							Requests: &armcontainerinstance.ResourceRequests{
+								CPU:        to.Ptr[float64](1),
+								MemoryInGB: to.Ptr[float64](1.5),
+							},
+						},
+					},
+				}},
+			IPAddress: &armcontainerinstance.IPAddress{
+				Type: to.Ptr(armcontainerinstance.ContainerGroupIPAddressTypePublic),
+				Ports: []*armcontainerinstance.Port{
+					{
+						Port:     to.Ptr[int32](3000),
+						Protocol: to.Ptr(armcontainerinstance.ContainerGroupNetworkProtocolTCP),
+					}},
+			},
+			OSType: to.Ptr(armcontainerinstance.OperatingSystemTypesLinux),
+			RestartPolicy: to.Ptr(armcontainerinstance.ContainerGroupRestartPolicyOnFailure),
+		},
+		Location: to.Ptr(containerGroupLocation),
+	}
+
+	poller, err := azureClient.NewContainerGroupsClient().BeginCreateOrUpdate(ctx, resourceGroupName, containerGroupName, containerGroup, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	// send request and wait until done
+	res, err := poller.PollUntilDone(ctx, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	// print ref
+	fmt.Println("Deployment for image", addr, "now available at", res.ContainerGroup.Properties.IPAddress.IP)
```

Comment:
This displays an object ID instead of the IP address. I'm not sure how to convert it to a string. I would like the otput to be `http://IP:3000`. Can you help @marcosnils @kpenfound ? The model details are in https://github.com/Azure/azure-sdk-for-go/blob/sdk/resourcemanager/containerinstance/armcontainerinstance/v2.3.0/sdk/resourcemanager/containerinstance/armcontainerinstance/models.go#L598

Reply:
Ping @marcosnils @kpenfound on this

Reply:
Since IP is a `*string` I think you just need to reference it:
```suggestion
	fmt.Println("Deployment for image", addr, "now available at", *res.ContainerGroup.Properties.IPAddress.IP)
```